### PR TITLE
REF: combine Block _can_hold_element methods

### PR DESCRIPTION
**Pull request by jbrockmendel**
which in turn lets us get rid of ObjectValuesExtensionBlock and HybridMixin.

### CODE REVIEW

File: pandas/core/dtypes/cast.py [hunk 1]
```diff
@@ -2212,4 +2251,11 @@ def can_hold_element(dtype: np.dtype, element: Any) -> bool:
     elif dtype == object:
         return True
 
+    elif dtype.kind == "S":
```
Comment (jreback) [inline pandas/core/dtypes/cast.py:2254]: do we ever have kind == 'U"?

Comment (jreback) [reply pandas/core/dtypes/cast.py:2254]: we should *never* have this actually (I think we are deprecating)?

File: pandas/tests/extension/test_numpy.py [hunk 2]
```diff
@@ -27,7 +28,10 @@
...
 import pandas._testing as tm
 from pandas.core.arrays.numpy_ import PandasArray
-from pandas.core.internals import managers
+from pandas.core.internals import (
+    blocks,
+    managers,
+)
 from pandas.tests.extension import base
 
 # TODO(ArrayManager) PandasArray
```
Comment (jreback) [inline pandas/tests/extension/test_numpy.py:37]: can you add a note here that these are for mocking (i know seems obvious, but a future reader may not be so clear)

**Review by jreback — CHANGES_REQUESTED** [review]
looks great a couple of quetions

File: pandas/core/dtypes/cast.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/dtypes/cast.py:2254]: my understanding is that this shouldnt be reached, but de facto it is and i decided fixing that was orthogonal to this PR

File: pandas/tests/extension/test_numpy.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/extension/test_numpy.py:37]: sure, ok for follow-up?  (CI is a few minutes from green)

**Review by jreback — APPROVED** [review]

**Comment by jbrockmendel** [comment]
Booyah!  #40614 gets rid of two more Block subclasses, then #40149 gets us within spitting distance of another after that

**Pull request merged**